### PR TITLE
Use a mutex/lock on the validator store

### PR DESCRIPTION
**Pull request by terenc3t**
Maps are not thread safe and this validator store operations may be called from many places.

### CODE REVIEW

**Comment by terenc3t** [comment]
Don't merge yet please. Need to test run time

File: beacon-chain/core/validators/validator.go [hunk 1]
```diff
@@ -351,7 +359,10 @@ func UpdateRegistry(ctx context.Context, state *pb.BeaconState) (*pb.BeaconState
 				break
 			}
 			state = ExitValidator(state, uint64(idx))
-			ExitedValidators[updatedEpoch] = append(ExitedValidators[updatedEpoch], uint64(idx))
+			vStore.Lock()
```
Comment (prestonvanloon) [inline beacon-chain/core/validators/validator.go]: Does this work? I think you are still in a lock from line 326

Comment (terenc3t) [reply beacon-chain/core/validators/validator.go]: Ah yes. Fixed.

**Comment by terenc3t** [comment]
Verified working run time. Going to merge this in.

**Pull request merged**